apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2.java

```diff
@@ -892,12 +947,50 @@ private void addC4DicIdentifier(Coverage coverage, Beneficiary beneficiary) {
                     null,
                     TransformerConstants.PATIENT_MB_ID_DISPLAY,
                     "MB"))
-            .setValue((String.valueOf(beneficiary.getBeneficiaryId())))
-            .setSystem(TransformerConstants.CODING_BBAPI_BENE_ID)
+            .setValue((String.valueOf(beneficiary.getMedicareBeneficiaryId())))
+            .setSystem(TransformerConstants.CODING_BBAPI_MEDICARE_BENEFICIARY_ID_UNHASHED)
             .setAssigner(new Reference(organization));
     coverage.addIdentifier(identifier);
   }
 
+  /**
+   * Denotes which parts the beneficiary is subscribed to on the provided {@link Coverage} resource.
+   *
+   * @param beneficiary the value for {@link Beneficiary}
+   * @param coverage The {@link Coverage} to Coverage details
+   */
+  private void addSubscribedParts(Beneficiary beneficiary, Coverage coverage) {
+    if (beneficiary.getPartACoverageStartDate().isPresent()) {
+      createCoverageClass(
+          coverage,
+          CoverageClass.PLAN,
+          TransformerConstants.COVERAGE_PLAN_PART_A,
+          Optional.empty());
+    }
+    if (beneficiary.getPartBCoverageStartDate().isPresent()) {
+      createCoverageClass(
+          coverage,
+          CoverageClass.PLAN,
+          TransformerConstants.COVERAGE_PLAN_PART_B,
+          Optional.empty());
+    }
+    if (beneficiary.getHmoCoverageCount().isPresent()
```

Comment:
Good find 👍. I was afraid we'd have to check each of the HMO monthly indicator columns.

Reply:
One potential question though - it looks like this field is updated gradually as the year goes on. If you look for a bene with reference year = 2024 and HMO coverage count = 8, there are no results, so it looks like this field is a reflection of the current year up until the previous month. If we query this in January, will we get a value of 0 or 1?

Reply:
This might just be an edge case we have to live with, but it would be good to know more about the behavior here.